pkg/aws/actuator/actuator_test.go

```diff
@@ -572,3 +573,165 @@ func (a *testAWSError) OrigErr() error {
 func (a *testAWSError) Error() string {
 	panic("not implemented")
 }
+
+func TestDetectSTS(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	codec, err := minterv1.NewCodec()
+	if err != nil {
+		t.Fatalf("failed to set up codec for tests: %v", err)
+	}
+
+	tests := []struct {
+		name               string
+		existing           []runtime.Object
+		wantErr            assert.ErrorAssertionFunc
+		ctx                context.Context
```

Comment:
nit: in tests this is always going to be `context.Background()` so perhaps not necessary to parameterize?

Reply:
yep, removing in upcoming commit